src/components/tx/SignOrExecuteForm/hooks.ts

```diff
@@ -32,6 +33,25 @@ type TxActions = {
     isRelayed?: boolean,
   ) => Promise<string>
   signDelegateTx: (safeTx?: SafeTransaction) => Promise<string>
+  proposeTx: (safeTx: SafeTransaction, txId?: string, origin?: string) => Promise<TransactionDetails>
+}
+
+type txDetails = AsyncResult<TransactionDetails>
+
+export const useProposeTx = (safeTx?: SafeTransaction, txId?: string, origin?: string): txDetails => {
+  const { proposeTx } = useTxActions()
+  const { safe } = useSafeInfo()
+
+  return useAsync(
+    () => {
+      if (txId) return getTransactionDetails(safe.chainId, txId)
+      if (!safeTx) return
```

Comment:
If we switch these lines we make fewer unnecessary tx details requests

Reply:
No, it's better to keep it this way. TxDetails can be fetched by txId even if safeTx is not ready yet.